src/api/implementation/cluster-explorer/v1.ts

```diff
@@ -103,7 +103,7 @@ function adaptKubernetesExplorerNode(node: ClusterExplorerNode): ClusterExplorer
             return { nodeType: 'helm.release', name: node.releaseName };
         case 'helm.history':
             return { nodeType: 'helm.history', name: node.releaseName };
-        case 'extension':
+        default:
```

Comment:
I think i edited this bc i was facing a typescript error `Function lacks ending return statement and return type does not include 'undefined'`. As it is now, the switch misses the default case. 
In v1_1 we have `default:  return { nodeType: 'extension' };` so i guess i copied from that. 
Adding it with a `default: return undefined`. Ok?

Reply:
Yeah, I agree you need to return the `extension` in old versions of the API, because we have no other way of expressing "we hadn't thought if this when we wrote this version of the API."  I was more thinking that in v1.2 (or v2 or whatever this turns out to be) we could add a new "unknown" type that would be available when _in a future release_ we added new node types that didn't match any of the 1.2 API types.

Reply:
By looking at it better i got what you were trying to say. Yes, the "unknown" node could be a life saver in future.
I noticed this issue while rebasing the `common.ts`. If someone is using the v1 or v1_1 api, we have no way to adapt the `ClusterExplorerNode` to those versions because there is the new namespace node that didn't exist before. The only way we have now is to return undefined which is not completely correct.

Reply:
Yeah, for backward compatibility v1 and v1,1 are stuck with using `extension` for untranslatable node types,  I suggest we add a new API node type as part of v1.2 (for future use) and just accept that 1 and 1,1 are stuck with the wart.

This makes `common.ts` a bit trickier; in fact I probably factored it out prematurely.  But I think I have a possible way forward at https://github.com/itowlson/vscode-kubernetes-tools/tree/versioning-woes-playground (note: this is NOT intended as a real 1.2 we should ship, just one I created to examine the issues we would face).  I'd value your feedback on that, or on any other strategies you have that will save us going back to full duplicated code.

Reply:
I gave a look at your branch. I like the way you solved the issue. Smart. :rocket: 
And no, i don't have any other strategy in mind. While rebasing i was thinking at something similar to what you did with `tryAdaptKubernetesExplorerNode1_2` and `tryAdaptKubernetesExplorerNode1_to_1_1` so it's ok for me